### ISSUE TITLE
Toolbar - Expose and divide Sculpting brushes into panels

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#1395]
# Issue
Right now, the sculpting brushes are many and fixed. You can't re-order nor hide any of the brushes - they all just.. bam, hit you in the face. 

# Solution
Expose the panel code, make new sculpting panel duplicates, then hide/remove duplicates to then create a new organization of panels for the brushes in their groups (not just colour groups). That is just a workflow to get it done. 

![bforartists_2020-03-05_10-35-25](https://user-images.githubusercontent.com/35357409/75997426-4d83a080-5ecd-11ea-9169-8a06e9dbd84b.png)

TLDR:
Expose and organize panels for the brush groups

### Reason
- It makes brushes collapsable for streamlining
- It shows re-ordering for artist preference
- It shows a title for the brush group for easier comprehension/learning of the brush types

**Comment by ReinerBforartists** [comment]
Nice dream that already turned out to be unmanageable. Unfortunately. So as sorry as i am, i have to close this one.

**Comment by Draise14** [comment]
Unmanageable? What do you mean? Re-organizing these panels and making new ones is easy, no errors. But adding new tools that aren't active and tabs is what's being difficult. I don't get errors when I do it. I could do this task technically. Why is it unmanageable?

**Comment by Draise14** [comment]
Ah, I see, it's a menu that's calling an enumerated list of brushes. Where this list is set, I dunno. I'll have to keep experimenting/exploring. For obvious tool lists that aren't enumerated, then yes.. you can split and organize.. but not these enumerated ones (there are 10 lists?!)

![atom_2020-03-05_16-37-12](https://user-images.githubusercontent.com/35357409/76027972-948a8980-5eff-11ea-9645-792521887efa.png)

**Comment by Draise14** [comment]
Aha, found them. 

`H:\bforartists-repository\Bforartists\source\blender\makesrna\intern\rna_brush.c`
![atom_2020-03-05_16-56-21](https://user-images.githubusercontent.com/35357409/76029357-4925aa80-5f02-11ea-9fb8-e807fa7e9ca6.png)

`H:\bforartists-repository\Bforartists\source\blender\makesdna\DNA_brush_types.h`
![atom_2020-03-05_17-01-34](https://user-images.githubusercontent.com/35357409/76029689-fdbfcc00-5f02-11ea-83c7-28d101dbd67c.png)

I can quickly see that yeah.. now I get why you think it's unmanageable. The way the brushes work is really all over the place, and enumeration makes life difficult as it's all compiled in C files in random places - and I still have no idea where or how they call these menus. 

![image](https://user-images.githubusercontent.com/35357409/76030765-58a5f300-5f04-11ea-9a46-a5ddafcefef8.png)

Technically I can see how to make new enumerated menus with the items with a dupilicate, and then call the new enumerated list, but where in what document might take some heavy digging...... 

Meh. What a chore. It's a great feature, but it's a chore to get. I see now.

**Comment by ReinerBforartists** [comment]
Yeah, this whole system is bad. Dividing into panels did not work. Tabs are impossible to place within the new toolsystem too. Even strings does not work.

Hm, i will reopen it. Maybe i can at least manage to group some of the brushes together at one point. But no promises.

**Comment by Draise14** [comment]
Hey, thanks for the open mind about it. I would like to prototype it personally at least.. The challenge is making another enumerated menu and then just listing that one in the same way this one is made, theoretically I think it would work - just not sure what loose ends I'll find till I give it a go. 

_It's fun diving into the code this deep, never actually have done it.  It's all interesting. It's so bureaucratic, there are like 2-4 layers of tool code in different documents before it even shows up in the UI..._

**Comment by ReinerBforartists** [comment]
Indeed. Hella complicated. That's why i usually avoid the C code at all ^^

**Comment by ReinerBforartists** [comment]
Had a final look. The tools enumeration is done in one go for all brushes. And honestly, i don't want to touch this part. It would quickly become a maintaining nightmare with all the changes that are still coming. We would also run into trouble with possible gaps when we drag the toolbar to three columns. We already have this problem with the experimental features. They create us a gap ...

No matter what we do, we run into trouble here. And so i think it's best to finally cancel this task. You can't always win. And this is such a case here where a change would make it worse.

![gaps](https://user-images.githubusercontent.com/13919702/115854306-6715dd80-a42a-11eb-93fe-b87c60d0ce5f.jpg)

**Comment by Draise14** [comment]
Yah, understandable. Mitigating to 3 columns kinda helps.
